src/ui/progress.js

```diff
@@ -90,36 +90,37 @@ class CanvasRenderer extends BaseRenderer {
   }
 
   update() {
-    var arc, ctx, half, size
-    half = Math.floor(Math.min(this.element.width(), this.element.height()))
-    size = half * 2
-    if (half) {
-      if (this.canvasEl.width !== size || this.canvasEl.height !== size) {
-        this.canvasEl.width = size
-        this.canvasEl.height = size
+    window.requestAnimationFrame(() => {
```

Comment:
Sync canvas render function somehow causes safari to do wrong menu height calculation if we trying to get it's child dimensions inside sync render function.

I've wrapped render function into raf, it's fixes this strange safari bug and just a better way to do animation loop.